lib/Yield/PTML.cpp

```diff
@@ -132,28 +163,36 @@ static std::set<std::string> targets(const yield::BasicBlock &BasicBlock,
     .OptionalCallAddress = MetaAddress::invalid()
   };
 
+  std::string NextBlock = targetPath(BasicBlock.nextBlock(), Function, Binary);
+
   std::set<std::string> Result;
   for (const auto &Edge : BasicBlock.Successors()) {
     auto TargetPair = efa::parseSuccessor(*Edge,
                                           BasicBlock.nextBlock(),
                                           Binary);
-    if (TargetPair.NextInstructionAddress.isValid()) {
+    if (TargetPair.NextInstructionAddress.isValid()
+        && TargetPair.NextInstructionAddress != BasicBlock.nextBlock()) {
       std::string Path = targetPath(TargetPair.NextInstructionAddress,
                                     Function,
                                     Binary);
-      if (!Path.empty()) {
+      if (!Path.empty())
```

Comment:
This looks more like an assertion. Same below.

Reply:
Ah, right, since we already checked whether it's `valid`, we can just assert it. My bad.

Reply:
I'm having second thoughts on this one, too.

My initial assumption about indirect calls is not right. This `just ignore them` branch is taken on any target that's not a known basic-block / function. We could theoretically assert it, but I feel like this is close enough to the user that I'd rather ignore "that unknown entity" over asserting on it.

Reply:
I'm not sure what case you're trying to handle here. If `TargetPair.NextInstructionAddress` is valid, we must be able to produce a location.

Assertions >> FUD. Either you have a specific example you can articulate or assert so we'll hit it and see if it's legit or a bug.

Reply:
Status?

Reply:
Done.